File: Plugins/Aspose_Slides_Java_for_Ruby/lib/asposeslidesjava/Shapes/rotatingshapes.rb
module Asposeslidesjava
  module RotatingShapes
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/Shapes/'
                
        # Create an instance of Presentation class
        pres = Rjb::import('com.aspose.slides.Presentation').new

        # Get the first slide
        sld = pres.getSlides().get_Item(0)

        # Add autoshape of rectangle type
        shp = sld.getShapes().addAutoShape(Rjb::import('com.aspose.slides.ShapeType').Rectangle, 50, 150, 75, 150)

        # Rotate the shape to 90 degree
        shp.setRotation(90)

        # Write the presentation as a PPTX file 
        save_format = Rjb::import('com.aspose.slides.SaveFormat')
        pres.save(data_dir + "RectShpRot.pptx", save_format.Pptx)

        puts "Rotated shape, please check the output file."
    end
  end
end

File: Examples/src/main/java/com/aspose/slides/examples/Slides/Shapes/RotatingShapes.java
package com.aspose.slides.examples.Slides.Shapes;

import com.aspose.slides.IShape;
import com.aspose.slides.ISlide;
import com.aspose.slides.Presentation;
import com.aspose.slides.SaveFormat;
import com.aspose.slides.ShapeType;
import com.aspose.slides.examples.Utils;

public class RotatingShapes {

	public static void main(String[] args) {

		// The path to the documents directory.
		String dataDir = Utils.getDataDir(RotatingShapes.class);

		// Instantiate Presentation class that represents the PPTX
		Presentation pres = new Presentation();

		// Get the first slide
		ISlide sld = pres.getSlides().get_Item(0);

		// Add AutoShape of rectangle type
		IShape shp = sld.getShapes().addAutoShape(ShapeType.Rectangle, 50, 150, 75, 150);

		// Rotate the shape to 90 degree
		shp.setRotation(90);

		// Write the PPTX file to disk
		pres.save(dataDir + "RectShpRot.pptx", SaveFormat.Pptx);

	}

}
